packages/tools/rpi-eeprom/sources/rpi-eeprom-update

```diff
@@ -253,43 +268,67 @@ fileUpdate()
    echo "*** INSTALLING ${1} ***"
    [ -f "${1}" ] || die "\"${1}\" not found"
    applyUpdate "$1"
-   echo "File update completed"
+   echo "Bootloader EEPROM update complete. Please reboot."
 }
 
-checkUpdateInProgress()
+removePreviousUpdates()
 {
    findBootFS
-   if [ -f "${BOOTFS}/recovery.bin" ]; then
-cat <<EOF
-Found recovery.bin on boot/recovery parition.
-It looks as though the previous EEPROM update failed. This can happen
-if the power supply is disconnected during the EEPROM update. Running
-the EEPROM update again should resolve this or alternatively use the
-rescue image from https://www.raspberrypi.org/downloads/
-EOF
-      exit ${EXIT_PREVIOUS_UPDATE_FAILED}
-   fi
+   match="RECOVERY.[0-9][0-9][0-9]"
```

Comment:
It's a bit ugly but [Rr][Ee][Cc][Oo][Vv][Ee][Rr][Y].[0-9][0-9][0-9] should do the trick, I'll see if I can think of something cleaner that works with busybox

Reply:
Thanks. But wouldn't it be safer to keep this case-sensitive (ie. `-regex` not `-iregex`)?

Reply:
case-sensitive means that if the bootloader is ever updated to do long file-names properly then it will potentially leave the old recovery files to accumulate. Probably not a disaster since they are small and updates should be very rare unless the user is tweaking the bootconf in which case they can probably fix any stale files themselves.

It looks as though iname works on busybox find

busybox find "${BOOTFS}" -maxdepth 1 -type f -iname "recovery.*" -regex "${BOOTFS}/.*[0-9][0-9][0-9]"

Reply:
Thanks for the explanation.

Yes, the combination of `iname` and `regex` is working with busybox. Do you think the following regex might be better as it avoids matching the mount point (not sure if that also could be uppercase/lowercase/either) - it basically just accepts anything that matches the `find ...` so long as it ends with 3 digits:
```
rpi4:~ # BOOTFS=/flash
rpi4:~ # touch /flash/RECOVERY.000
rpi4:~ # touch /flash/RECOVERY.001
rpi4:~ # find "${BOOTFS}" -maxdepth 1 -type f -iname "recovery.*" -regex ".*\.[0-9][0-9][0-9]$"
/flash/RECOVERY.000
/flash/RECOVERY.001
```

Reply:
"ends with a 3 digit file extension (and only 3 digits)" - your original version could match `recovery.A123`

Reply:
good point, I'll pull that change in :)